App/Playlist.js

```diff
@@ -0,0 +1,57 @@
+function Playlist(data) {
+  this.index = data.id;
+  this.playlistName = data.title;
+  this.imageLarge = data.imageLarge;
+  this.imageSmall = data.imageSmall;
+  this.description = data.description;
+  this.songsList = [];
+  this.observers = [];
+}
+Playlist.prototype.setName = function(name){
+  this.playlistName = name;
+};
+Playlist.prototype.addSong = function(song){
+    this.songsList.push(song);
+    this.notify(song);
+};
+Playlist.prototype.addObserver = function(observer) {
+  this.observers.push(observer);
+};
+Playlist.prototype.notify = function(song){
+  for(var i = 0; i < this.observers.length; i++)
+    {
+      this.observers[i].drawUpdate(song);
```

Comment:
just call it update. Don't tie it to the action of drawing. Maybe an observer want to show a notification upon update etc